poly/sharing_test.go

```diff
@@ -0,0 +1,544 @@
+package poly
+
+import (
+	"testing"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/edwards"
+	"github.com/dedis/crypto/random"
+)
+
+/* This file is a testing suite for sharing.go. It provides multiple test cases
+ * for ensuring that encryption schemes built upon this package such as Shamir
+ * secret sharing are safe and secure.
+ *
+ * The tests can also serve as references for how to work with this library.
+ */
+
+/* Global Variables */
+
+var group abstract.Group = new(edwards.ExtendedCurve).Init(
+	edwards.Param25519(), false)
+var altGroup abstract.Group = new(edwards.ProjectiveCurve).Init(
+	edwards.ParamE382(), false)
+var k int = 10
+var n int = 20
+var secret = group.Secret().Pick(random.Stream)
+var point = group.Point().Mul(group.Point().Base(), secret)
+var altSecret = altGroup.Secret().Pick(random.Stream)
+var altPoint = altGroup.Point().Mul(altGroup.Point().Base(), altSecret)
+
+/* Setup Functions
+ *
+ * These functions provide greater modularity by consolidating commonly used
+ * setup tasks.
+ *
+ * Not every function uses these methods, since they may have unique set-up
+ * needs that do not warrant their own set-up function.
+ */
+
+// Tests that checks whether a method panics can use this funcition
+func deferTest(t *testing.T, message string) {
+	if r := recover(); r == nil {
+		t.Error(message)
+	}
+}
+
+func producePriPoly(g abstract.Group, k int, s abstract.Secret) *PriPoly {
+	return new(PriPoly).Pick(g, k, s, random.Stream)
+}
+
+func produceTwoPriPoly(g abstract.Group, k int, s abstract.Secret) (*PriPoly,
+	*PriPoly) {
+	return producePriPoly(g, k, s), producePriPoly(g, k, s)
+}
+
+func produceThreePriPoly(g abstract.Group, k int, s abstract.Secret) (*PriPoly,
```

Comment:
Do you really need three versions of this function?  How about just keeping producePriPoly and calling it 2 or 3 times in the few places where you need 2 or 3 polynomials?